File: py/solns/palindromicSubstrs/palindromicSubstrs.py
class Solution:
    def __init__(self):
        self.res = []
    
    def naive(self,s):
        self.length = len(s)
        def loop(start,end):
            l,r = start,end
            while l>=0 and r<=self.length-1 and s[l]==s[r]:
                self.res.append(s[l:r+1])
                l-=1
                r+=1
        for i in range(self.length):
            loop(i,i)
            loop(i,i+1)
        return len(self.res)

File: cpp/example/src/PalindromicSubstrs/palindromicSubstrs.cpp
#include "palindromicSubstrs.h"

void palindromicSubstrs::loop(int start,int end){
    int l=start,r=end;
    while (l>=0 && r<=this->length-1 && this->s[l]==this->s[r]){
        this->res+=1;
        l-=1;
        r+=1;
    }
}

int palindromicSubstrs::naive(string s){
    this->s = s;
    this->length = s.size();
    for (int i=0;i<this->length;i++){
        loop(i,i);
        loop(i,i+1);
    }
    return this->res;
}
